dask/dataframe/indexing.py

```diff
@@ -114,14 +114,18 @@ def _loc(self, iindexer, cindexer):
                 return self._loc_slice(iindexer, cindexer)
             elif isinstance(iindexer, (list, np.ndarray)):
                 return self._loc_list(iindexer, cindexer)
-            elif is_series_like(iindexer) and iindexer.dtype != bool:
+            elif is_series_like(iindexer) and iindexer.dtype not in [
+                bool,
+                pd.BooleanDtype(),
```

Comment:
This doesn't quite work for older versions of pandas. It is probably better to use `pandas.api.types.is_bool_dtype` instead of checking the type directly.

Reply:
Thanks for your advice, I was not aware of that function. I used the dask-variant you mentioned above everywhere.

Reply:
Oh, I overlooked that `.meta` there. I guess using it we prohibit may using the full array?